modules/Hoot/ui/managePanel/datasets.js

```diff
@@ -247,7 +247,7 @@ export default class Datasets extends Tab {
 
                 if ( !confirm ) return;
 
-                let items = d.data.type === 'folder' ? new Array( d ) : tree.selectedNodes;
+                let items = d.data.type === 'folder' && !d.data.selected ? new Array( d ) : tree.selectedNodes;
```

Comment:
let's put some parentheses here to capture the conditional part of the ternary.

Reply:
updated